dask/array/rechunk.py

```diff
@@ -205,7 +203,7 @@ def rechunk(x, chunks):
 
     crossed = intersect_chunks(x.chunks, chunks)
     x2 = dict()
-    temp_name = next(rechunk_names)
+    temp_name = 'rechunk-' + tokenize((x.name, chunks))
```

Comment:
Sometimes things need to be normalized, and sometimes they don't. If they don't, there's no reason to call an extra function on each argument. In this case there wouldn't be a big difference, but some functions have very long tuples of args, none of which need to be normalized.